src/viam/sdk/common/proto_utils.hpp

```diff
@@ -0,0 +1,61 @@
+/// @file common/proto_utils.hpp
+///
+/// @brief Utils that require generated proto includes. These should be #included
+/// in cpp implementation files, but not in wrapper headers consumed by third party code.
+#pragma once
+
+#include <viam/api/common/v1/common.pb.h>
+
+namespace viam {
+namespace sdk {
+
+/// @brief Copies elements from a protobuf repeated pointer array into a std::vector. Src type
+/// must be implicitly convertible to Dst (probably via operator on Src).
+template <typename Src, typename Dst>
+void vecToRepeatedPtr(const std::vector<Src>& vec, google::protobuf::RepeatedPtrField<Dst>& dest) {
```

Comment:
So generally speaking I'm very much against output parameters being passed as pointers, it's one of the things I dislike the most about google's C++ coding guidelines. I think with your counterexample above the point is that, in C++ and as high level as this code is, we generally aren't passing around pointer types in the first place, and the point of references is that they must point to a valid object or else it's UB. So as for

> oops, what if pdest is nullptr

That's on the caller for dereferencing a null pointer in the first place, and is equally true of if `my_vec` was `*p_my_vec` for some reason, and this is kind of the point of C++ having reference types. 

Callsite legibility I'm somewhat sympathetic to but would be better done with something like `std::ref(arg)` or `gsl::output_parameter`, because then you don't lose the non-null guarantee. 

.....All that said, I think this could be a pointer parameter just for ABI insulation reasons